service/matching/matcher.go

```diff
@@ -148,12 +246,27 @@ func (tm *TaskMatcher) Poll(ctx context.Context) (*internalTask, error) {
 	case <-ctx.Done():
 		tm.scope().IncCounter(metrics.PollTimeoutCounter)
 		return nil, ErrNoTasks
+	case token := <-tm.fwdrPollReqTokenC():
+		if task, err := tm.fwdr.ForwardPoll(ctx); err == nil {
```

Comment:
before actually polling from root / remote, maybe another attempt of polling from `tm.taskC` and `tm.queryTaskC`?

Reply:
basically, i think polling from local should have higher priority, event if task manager has token to poll from remote

Reply:
I did consider this and then decided to drop this idea for the first round and here is the reasoning:

Before entering the select{} block which has both local task channels and forwarderTokenC, we always first exclusively attempt to match only with local task channels i.e. we always prioritize local pollers in the first select{} block before entering the second. In a busy task list, the first select{} should almost always match against a local poller. On the other hand, when its not matched, the task list probably isn't all that busy and its ok to forward. Having said that, my plan is to observe the forwardedFrom / matching metrics from a real stress test and use that to decide if adding this additional nested select{} is worth it. Till then, I prefer to start simple.